frontend/src/components/design-system/input/ButtonComponent.vue

```diff
@@ -0,0 +1,270 @@
+    <span
+      v-if="$slots.icon"
+      :class="['icon', $slots.default ? 'icon--before' : 'icon--only']"
+    >
+      <slot name="icon" :size="computedSize" />
+    </span>
+    <ButtonText v-if="$slots.default" :size="computedSize">
+      <slot />
+    </ButtonText>
+    <span v-if="$slots.iconAfter" :class="['icon', 'icon--after']">
+      <slot name="iconAfter" :size="computedSize" />
+    </span>
+  </component>
+</template>
+
+<style scoped lang="scss">
+.button {
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  justify-content: space-between;
+  line-height: 0;
+  padding: 0;
+  cursor: pointer;
+  background-color: transparent;
+  text-decoration: none;
+
+  &:disabled {
+    cursor: not-allowed;
+  }
+
+  &--link {
+    color: rgb(var(--color-neutral-fg));
+    border: none;
+
+    &:hover {
+      color: rgb(var(--color-neutral-fg-hover));
+    }
+
+    .button--selected,
+    &:active {
+      color: rgb(var(--color-neutral-fg-active));
+    }
+
+    &:disabled {
+      color: rgb(var(--color-neutral-fg-disabled));
+      border-color: rgb(var(--color-neutral-fg-disabled));
+    }
+
+    .icon {
+      &--before {
+        margin-right: var(--spacing-extra-small);
+      }
+
+      &--after {
+        margin-left: var(--spacing-extra-small);
+      }
+    }
+  }
+
+  &--outlined,
+  &--filled,
+  &--answer {
+    border-style: solid;
+    border-width: var(--spacing-nano);
+    border-radius: var(--radius-small);
+    border-top-right-radius: 0;
+
+    &.button--medium {
+      padding: calc(var(--spacing-small) - var(--spacing-nano));
+    }
+
+    &.button--small {
+      padding-left: calc(var(--spacing-small) - var(--spacing-nano));
+      padding-right: calc(var(--spacing-small) - var(--spacing-nano));
+      padding-top: calc(var(--spacing-extra-small) - var(--spacing-nano));
+      padding-bottom: calc(var(--spacing-extra-small) - var(--spacing-nano));
+    }
+
+    .icon {
+      &--before {
+        margin-right: var(--spacing-small);
+      }
+
+      &--after {
+        margin-left: var(--spacing-small);
+      }
+    }
+  }
+
+  &--outlined {
+    background-color: transparent;
+    border-color: rgb(var(--color-neutral-border-strong));
+
+    &:hover {
+      background-color: rgba(
+        var(--color-neutral-fg-hover),
+        var(--transparency-10)
+      );
+    }
+
+    .button--selected,
+    &:active {
+      background-color: rgba(
+        var(--color-neutral-fg-active),
+        var(--transparency-10)
+      );
+    }
+
+    &:disabled {
+      background-color: transparent;
+      color: rgb(var(--color-neutral-fg-disabled));
+      border-color: rgb(var(--color-neutral-border-disabled));
+    }
+  }
+
+  &--filled,
+  &--answer {
+    &:disabled {
+      color: rgb(var(--color-primary-fg-inverse)) !important;
+      background-color: rgb(var(--color-neutral-bg-disabled)) !important;
+      border-color: transparent !important;
```

Comment:
Pokud někde nemáš overriding, kterýho jsem si nevšiml, tak `!important` není potřeba. Class a pseudoselektor je vždy víc než jen classa samotná. Navíc to máš jen zde.

Reply:
Je, přepisují to níže ty typy tlačítek, kde to má vyšší specificitu.